custom/icds_reports/utils/data_pull.sql

```diff
@@ -0,0 +1,117 @@
+--                                  Sort Key: child_health_monthly.caste
+--                                  ->  Partial HashAggregate  (cost=74350.90..74352.90 rows=200 width=12)
+--                                        Group Key: child_health_monthly.caste
+--                                        ->  Parallel Append  (cost=0.00..74244.24 rows=21332 width=4)
+--                                              ->  Parallel Seq Scan on "child_health_monthly_2020-05-01_671996" child_health_monthly  (cost=0.00..74137.58 rows=21332 width=4)
+--                                                    Filter: ((month = '2020-05-01'::date) AND (valid_in_month = 1) AND (alive_in_month = 1) AND (state_id = 'f9b47ea2ee2d8a02acddeeb491d3e175'::text) AND (age_tranche = ANY ('{0,6,12,24,36}'::text[])))
+
+SELECT caste, count(*) as number_3_6_child FROM "child_health_monthly" WHERE
+    month='2020-05-01'
+    AND age_tranche IN ('48', '60', '72')
+    AND valid_in_month=1
+    AND alive_in_month=1
+    AND state_id='f9b47ea2ee2d8a02acddeeb491d3e175'
+    GROUP BY caste;
+
+-- QUERY PLAN
+-- -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+--  HashAggregate  (cost=0.00..0.00 rows=0 width=0)
+--    Group Key: remote_scan.caste
+--    ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+--          Task Count: 64
+--          Tasks Shown: One of 64
+--          ->  Task
+--                Node: host=100.71.184.232 port=6432 dbname=icds_ucr
+--                ->  Finalize GroupAggregate  (cost=74808.50..74910.29 rows=200 width=12)
+--                      Group Key: child_health_monthly.caste
+--                      ->  Gather Merge  (cost=74808.50..74904.29 rows=800 width=12)
+--                            Workers Planned: 4
+--                            ->  Sort  (cost=73808.44..73808.94 rows=200 width=12)
+--                                  Sort Key: child_health_monthly.caste
+--                                  ->  Partial HashAggregate  (cost=73798.80..73800.80 rows=200 width=12)
+--                                        Group Key: child_health_monthly.caste
+--                                        ->  Parallel Append  (cost=0.00..73679.12 rows=23937 width=4)
+--                                              ->  Parallel Seq Scan on "child_health_monthly_2020-05-01_671996" child_health_monthly  (cost=0.00..73559.43 rows=23937 width=4)
+--                                                    Filter: ((month = '2020-05-01'::date) AND (valid_in_month = 1) AND (alive_in_month = 1) AND (state_id = 'f9b47ea2ee2d8a02acddeeb491d3e175'::text) AND (age_tranche = ANY ('{48,60,72}'::text[])))
+
+SELECT ccs.caste, count(*) as number_pw FROM "ccs_record_monthly" ccs
+    LEFT JOIN "awc_location" awc ON awc.doc_id = ccs.awc_id
```

Comment:
you will need supervisor_id in the join condition, both to make this query work and to work as distributed.